sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ReviewSqlQueriesForSecurityVulnerabilities.cs

```diff
@@ -85,11 +114,90 @@ protected override void Initialize(SonarAnalysisContext context)
                         c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, assignment.Left.GetLocation()));
                     }
                 }, SyntaxKind.SimpleAssignmentExpression);
+
+            context.RegisterCompilationStartAction(
+                cc =>
+                {
+                    if (!TryReadConfiguration(cc.Options))
+                    {
+                        return;
+                    }
+
+                    protobufFileIndex = 0;
+
+                    cc.RegisterSyntaxNodeActionInNonGenerated(
```

Comment:
Out of interest, why did you choose to register for callbacks rather than just walking the tree?

Reply:
Usually we prefer callbacks because you don't need to create a new walker class. I personally choose walker when some state has to be kept between walking the different syntax nodes.